torch/ao/quantization/fx/_model_report/model_report.py

```diff
@@ -111,9 +111,9 @@ def get_observers_of_interest(self) -> Dict[str, Set[str]]:
         """ Returns a copy of the observers of interest for viewing """
         return self._detector_name_to_observer_fqns.copy()
 
-    def prepare_detailed_calibration(self, prepared_fx_model: GraphModule) -> GraphModule:
+    def prepare_detailed_calibration(self, model: GraphModule) -> GraphModule:
```

Comment:
this still assumes the model is symbolically traced, I think ideally we should update the type to `nn.Module` for top level apis and do tracing inside the function

Reply:
I think we can have a further discussion about this, but i feel like doing this would make it harder to integrate this into quantization workflow. Specifically, how would we handle the flow:

Initialize model &rarr; Prepare model &rarr; Callibrate model &rarr; Convert model &rarr; ...

that we typically find in quantization. If we were to do the tracing inside the function, then that conflicts with what the `prepare` step is doing in the process.

So I think i am still confused how / why we would want to do tracing within ModelReport

Reply:
I'm not sure why do we have to use this inside quantization flow, this sounds something that is independent of quantization flow itself, maybe we can align on that first